### ISSUE TITLE
bug: sendrawtransaction discrepancy

### ISSUE THREAD

**Issue by conradoplg** [ZcashFoundation/zebra#9056]
### What happened?

 I expected to see this happen: Ywallet hasn't updated its consensus branch ID. When submitting a tx via `sendrawtransaction`, zcashd returns

```
error code: -26
error message:
16: bad-tx-consensus-branch-id-mismatch
```

Instead, this happened: zebra returned a transaction ID. However, later calls to `getrawmempool` did not list that txid. Trying to submit the tx again returns 

```
error code: 0
error message:
any transaction with the same effects will be rejected from the mempool until the next chain tip block
```

Look into it to check if it's just an error handling issue or if there is any bigger issue there.

(TODO: check my logs to see if there are more details)

### Which operating systems does the issue happen on?

- [ ] Linux
- [ ] macOS
- [ ] Windows
- [ ] Other OS

**Comment by mpguerra** [comment]
Related to #9063 Need to check that this is still happening after it merges

**Comment by mpguerra** [comment]
@conradoplg is it possible to test this now that #9064 is closed? Otherwise, if you can confirm that this is the same issue, we can close this one also

**Comment by conradoplg** [comment]
I didn't manage to test this today, but I'm pretty sure it was fixed by #9064. I think we can close this, but I can test it later if you prefer.

**Comment by mpguerra** [comment]
Let's close for now... maybe we can add it to the list of tests for NU7
